simulator/simulator.go

```diff
@@ -65,10 +57,22 @@ func startSimulator() error {
 		return xerrors.Errorf("create an etcd client: %w", err)
 	}
 
-	// need to sleep here to make all controllers create initial resources. (like "system-" priorityclass.)
-	time.Sleep(1 * time.Second)
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+	defer cancel()
+	err = wait.PollUntilContextCancel(ctx, time.Second*5, true, func(ctx context.Context) (bool, error) {
+		_, err := client.CoreV1().Namespaces().Get(context.Background(), "kube-system", metav1.GetOptions{})
+		if err != nil {
+			klog.Infof("waiting for kube-system namespace to be ready: %v", err)
+			return false, err
```

Comment:
I guess if we here return err, it'd immediately fail without retry. Let's just return nill as we log error in L65.

```suggestion
			return false, nil
```